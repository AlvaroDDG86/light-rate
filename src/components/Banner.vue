<template>
  <div
    class="
      banner
      flex
      inline-block
      bg-blue-700
      text-gray-100
      overflow-hidden
      h-20
      whitespace-nowrap
      text-center
      relative
    "
  >
    <span class="text mb-auto mt-auto text-5xl uppercase absolute overflow-hidden">
      {{ message }}
    </span>
  </div>
</template>
<script>
export default {
  props: {
    message: {
      type: String,
      required: true,
      default: "",
    },
  },
}
</script>
<style lang="scss" scoped>
.banner {
    justify-content: center;
    
    @media (max-width: 1240px) {
      justify-content: inherit;
    }
}
.text {
    top: 10%;
    animation: linear infinite;
    animation-duration: 10s;

    @media (max-width: 1240px) {
      justify-content: inherit;
      animation-name: run;
    }

    @media (max-width: 800px) {
      animation-name: run-m;
    }
   
    @media (max-width: 450px) {
      animation-name: run-xs;
    }
}

@keyframes run {
  from { margin-left: 100%; }
  to { margin-left: -100%; } 
}
@keyframes run-m {
  from { margin-left: 100%; }
  to { margin-left: -150%; } 
}
@keyframes run-xs {
  from { margin-left: 100%; }
  to { margin-left: -300%; } 
}
</style>