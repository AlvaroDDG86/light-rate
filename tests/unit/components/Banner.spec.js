import { shallowMount } from '@vue/test-utils'
import Banner from '@/components/Banner'

describe('Hour', () => {
    let wrapper
    beforeEach(() => {
        wrapper = shallowMount(Banner, {
            props: {
                message: 'Meesage example'
            }
        })
    })
    test('should match with snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot()
    });
    test('should render the message', () => {
        setTimeout(() => {
            const message = wrapper.find('span')
            expect(message.text()).toBe('Meesage example')
        }, 1000);
    });
});