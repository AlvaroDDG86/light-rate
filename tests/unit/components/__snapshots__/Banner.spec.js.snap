// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hour should match with snapshot 1`] = `
<div class="
    banner
    flex
    inline-block
    bg-blue-700
    text-gray-100
    overflow-hidden
    h-20
    whitespace-nowrap
    text-center
    relative
  "><span class="text mb-auto mt-auto text-5xl uppercase absolute overflow-hidden">

  </span></div>
`;
